tests/cps-files/lib/cps/full.cps

```diff
@@ -0,0 +1,53 @@
+{
```

Comment:
@mwoehlke Does this look like a reasonable and "full" example?

Reply:
What is "full"? Is this just supposed to be "complete", or is the goal to test as much as possible? Because if the latter, there's a lot missing, even if we ignore all the Platform stuff (which is reasonable to do for now, since no one is really satisfied with its current state).

I notice this is CPS 0.10. But the specification is on 0.12.

I haven't looked at the entire test suite, but things that should probably be tested that this doesn't test:
- Both language-specific and "global" includes, defines, etc.
- Paths that use `@prefix@`
- Multi-file stuff
- Requirements (both package and component)
- `compile_features`
- `link_languages`

In any case, I'd be tempted to make this `version: 1.2.1` and add `compat_version: 1.0.0`.

Reply:
I should have better explained this, but JSON doesn't allow comments so it's kind of awkward to do so. I intended `full.cps` to be a basic test case for every feature. Like it should have both `L` flags and `l` flags so that things like `--libs-only-L` can be tested. It should have multiple components so that `--component` can be tested. And so on. I imagine there to be generally two types of test cases, a test case for each functionality to verify it's sane, and edge test cases that keep growing as we discover those.

That said, it's hardly "full" now, but I expect it to be the dumping site of new stuffs as we add more functionalities. I agree tho the name is not great. I'd be open to a better name.